### PR TITLE
feat:[CDS-92168]: Support for Gitx in override CRUD api and import APIS

### PR DESCRIPTION
**Pull request by sourabh-awashti**
## Describe your changes
We have updated service override get, create, update APIs to support Gitx for remote service overrides.
Added the new import API to import remote override to harness.

Tests Done:
1. Creation, updation for all types of overrides at Account/org/project level for through Gitx (Remote overrides).
2. Creation, updation for all types of overrides at Account/org/project level for inlines override 
3. Import of remote overrides for all types at Account/Org/porject level.
4. Sanity validation with all existing overrides test cases.

### Import From Git
Screenshot 2024-03-01 at 11 47 43 AM

### Remote and Inline Service overrides
Screenshot 2024-02-29 at 6 43 54 PM
Screenshot 2024-02-29 at 6 42 06 PM

### Remote Service override update
Screenshot 2024-02-29 at 6 08 31 PM

### Inline Override Creation
Screenshot 2024-02-29 at 6 10 36 PM

### Inline Override Updation
Screenshot 2024-02-29 at 6 11 12 PM
## Comment Triggers

<details>
  <summary>PR Check triggers</summary>
  
- Build: `trigger build`
- Sub Category Field Check: `trigger subcategoryfieldcheck`
- gitleaks: `trigger gitleaks`

### CODE REVIEW

**Review by lovish1999 — APPROVED** [review]

File: internal/service/platform/service_overrides_v2/data_source_service_overrides_v2.go [hunk 1]
```diff
@@ -53,6 +53,35 @@ func DataSourceServiceOverrides() *schema.Resource {
 				ForceNew:    true,
 				Required:    true,
 			},
+			"git_details": {
+				Description: "Contains parameters related to Git Experience for remote entities",
```
Comment (hingerabhinav) [inline internal/service/platform/service_overrides_v2/data_source_service_overrides_v2.go:57]: overrides

File: internal/service/platform/service_overrides_v2/data_source_service_overrides_v2.go [hunk 2]
```diff
@@ -53,6 +53,35 @@ func DataSourceServiceOverrides() *schema.Resource {
...
+				Computed:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"branch": {
+							Description: "Name of the branch.",
+							Type:        schema.TypeString,
+							Optional:    true,
+							Computed:    true,
+						},
+						"load_from_fallback_branch": {
```
Comment (hingerabhinav) [inline internal/service/platform/service_overrides_v2/data_source_service_overrides_v2.go:70]: is load from cache param needed to be exposed to the end user?

File: internal/service/platform/service_overrides_v2/data_source_service_overrides_v2.go [hunk 3]
```diff
@@ -53,6 +53,35 @@ func DataSourceServiceOverrides() *schema.Resource {
...
+							Description: "Load service yaml from fallback branch",
+							Type:        schema.TypeBool,
+							Optional:    true,
+							Computed:    true,
+						},
+						"repo_name": {
+							Description: "Repo name of remote service",
```
Comment (hingerabhinav) [inline internal/service/platform/service_overrides_v2/data_source_service_overrides_v2.go:77]: overrides

File: docs/resources/platform_service_overrides_v2.md [hunk 4]
```diff
@@ -77,6 +142,23 @@ resource "harness_platform_service_overrides_v2" "test" {
...
+
+<a id="nestedblock--git_details"></a>
+### Nested Schema for `git_details`
+
+Optional:
+
+- `branch_name` (String) Name of the branch.
+- `connector_ref` (String) Identifier of the Harness Connector used for importing entity from Git To reference a connector at the organization scope, prefix 'org' to the expression: org.{identifier}. To reference a connector at the account scope, prefix 'account` to the expression: account.{identifier}.
+- `file_path` (String) File path of the Entity in the repository.
+- `repo_name` (String) Name of the repository.
```
Comment (hingerabhinav) [inline docs/resources/platform_service_overrides_v2.md:154]: nit: lets add the restriction of repo to be belong to the correct parent entity service or env.

File: internal/service/platform/service_overrides_v2/resource_service_overrides_v2.go [hunk 5]
```diff
@@ -54,14 +55,117 @@ func ResourceServiceOverrides() *schema.Resource {
...
 				Computed:    true,
 			},
+			"is_force_import": {
+				Description: "force import service from remote even if same file path already exist",
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Computed:    true,
+			},
+			"import_from_git": {
+				Description: "import service from git",
```
Comment (hingerabhinav) [inline internal/service/platform/service_overrides_v2/resource_service_overrides_v2.go:74]: overrides

File: internal/service/platform/service_overrides_v2/resource_service_overrides_v2_test.go [hunk 6]
```diff
@@ -26,8 +26,6 @@ func TestAccServiceOverrides_ProjectScope(t *testing.T) {
 			{
 				Config: testAccServiceOverridesProjectScope(id, name),
 				Check: resource.ComposeTestCheckFunc(
-					resource.TestCheckResourceAttr(resourceName, "org_id", id),
-					resource.TestCheckResourceAttr(resourceName, "project_id", id),
```
Comment (hingerabhinav) [inline internal/service/platform/service_overrides_v2/resource_service_overrides_v2_test.go:30]: we can add this back

**Review by hingerabhinav — APPROVED** [review]

**Pull request merged**